core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1530,15 +1531,28 @@ void renameInternal(LockedInodePath srcInodePath, LockedInodePath dstInodePath,
       String ufsSrcUri = resolution.getUri().toString();
       UnderFileSystem ufs = resolution.getUfs();
       String ufsDstUri = mMountTable.resolve(dstPath).getUri().toString();
-      String parentUri = new AlluxioURI(ufsDstUri).getParent().toString();
-      if (!ufs.exists(parentUri)) {
-        Permission parentPerm = new Permission(srcParentInode.getOwner(), srcParentInode.getGroup(),
-            srcParentInode.getMode());
-        MkdirsOptions parentMkdirsOptions = new MkdirsOptions().setCreateParent(true)
-            .setPermission(parentPerm);
-        if (!ufs.mkdirs(parentUri, parentMkdirsOptions)) {
-          throw new IOException(ExceptionMessage.FAILED_UFS_CREATE.getMessage(parentUri));
+      // Create ancestor directories from top to the bottom. We cannot use recursive create parents
```

Comment:
Hmm, current implementation in two places are different, one is getting permission from inode list, another one is getting permission by `getStatus()`. I think later on if we identified more places for this create ancestor directories logic, we should consolidate it to a until function.